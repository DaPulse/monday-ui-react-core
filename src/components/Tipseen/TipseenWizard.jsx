import React, { useMemo } from "react";
import cx from "classnames";
import Steps from "components/Steps/Steps";
import Button from "components/Button/Button";
import { BEMClass } from "helpers/bem-helper";
import TipseenBasicContent from "./TipseenBasicContent";
import "./TipseenWizard.scss";

const BASE_CSS_CLASS = "monday-style-tipseen-wizard";
const bemHelper = BEMClass(BASE_CSS_CLASS);

const TipseenWizard = ({ title, className, ...stepsProps }) => {
  const nextButtonProps = useMemo(
    () => ({
      kind: Button.kinds.PRIMARY,
      size: Button.sizes.SMALL
    }),
    []
  );
  const backButtonProps = useMemo(
    () => ({
      size: Button.sizes.SMALL
    }),
    []
  );
  return (
    <TipseenBasicContent title={title} className={cx(BASE_CSS_CLASS, className)}>
      <Steps
        className={bemHelper({ element: "wizard" })}
        isOnPrimary
        isContentOnTop
        areButtonsIconsHidden
        backButtonProps={backButtonProps}
        nextButtonProps={nextButtonProps}
        {...stepsProps}
      />
    </TipseenBasicContent>
  );
};
export default TipseenWizard;
