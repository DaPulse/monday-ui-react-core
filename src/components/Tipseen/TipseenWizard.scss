@import "./TipseenContent";
.monday-style-tipseen-wizard {
  display: flex;
  flex-direction: column;

  &_wizard {
    width: 100%;
  }
}