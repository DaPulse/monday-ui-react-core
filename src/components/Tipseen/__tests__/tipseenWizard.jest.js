import React from "react";
import renderer from "react-test-renderer";
import TipseenWizard from "../TipseenWizard";

const steps = [
  <div>Popover message will appear here loremipsum dolor samet…</div>,
  <div>Popover message will appear here loremipsum dolor samet…</div>,
  <div>Popover message will appear here loremipsum dolor samet…</div>
];

jest.useFakeTimers();
const renderComponent = ({ ...props }) => {
  return render(
    <TipseenWizard steps={steps} {...props}>
      <div />
    </TipseenWizard>
  );
};

describe("Tipseen wizard tests", () => {
  describe("Snapshot Tests", () => {
    it("renders correctly", () => {
      const tree = renderer
        .create(
          <TipseenWizard title="title" steps={steps} className="mock-class-name">
            <div />
          </TipseenWizard>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    it("renders correctly with empty props", () => {
      const tree = renderer.create(<TipseenWizard />).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it("renders correctly without title", () => {
      const tree = renderer
        .create(
          <TipseenWizard steps={steps}>
            <div />
          </TipseenWizard>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
