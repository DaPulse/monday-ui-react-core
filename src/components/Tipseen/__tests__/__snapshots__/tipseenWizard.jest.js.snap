// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tipseen wizard tests Snapshot Tests renders correctly 1`] = `
<div
  className="monday-style-tipseen-basic-content monday-style-tipseen-wizard mock-class-name"
>
  <span
    aria-level="3"
    className="monday-style-tipseen-title"
    role="heading"
  >
    title
  </span>
  <div
    className="monday-style-steps monday-style-tipseen-wizard_wizard monday-style-steps--on-primary monday-style-steps--content-on-top"
    id=""
  >
    <div>
      Popover message will appear here loremipsum dolor samet…
    </div>
    <div
      className="monday-style-steps-header monday-style-steps-header--gallery"
    >
      <button
        aria-disabled={true}
        className="monday-style-steps-command monday-style-steps-command--backward monday-style-button monday-style-button--size-small monday-style-button--kind-tertiary monday-style-button--color-on-primary-color"
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        type="button"
      >
        Back
      </button>
      <div
        className="monday-style-steps-header_gallery"
        role="group"
      >
        <button
          aria-current="step"
          aria-label="Step number 0"
          className="monday-style-steps-header_dot monday-style-steps-header_dot--is-active"
          onClick={[Function]}
          type="button"
        />
        <button
          aria-current={false}
          aria-label="Step number 1"
          className="monday-style-steps-header_dot"
          onClick={[Function]}
          type="button"
        />
        <button
          aria-current={false}
          aria-label="Step number 2"
          className="monday-style-steps-header_dot"
          onClick={[Function]}
          type="button"
        />
      </div>
      <button
        aria-disabled={false}
        className="monday-style-steps-command monday-style-steps-command--forward monday-style-button monday-style-button--size-small monday-style-button--kind-primary monday-style-button--color-on-primary-color"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        type="button"
      >
        Next
      </button>
    </div>
  </div>
</div>
`;

exports[`Tipseen wizard tests Snapshot Tests renders correctly with empty props 1`] = `
<div
  className="monday-style-tipseen-basic-content monday-style-tipseen-wizard"
>
  <div
    className="monday-style-steps monday-style-tipseen-wizard_wizard monday-style-steps--on-primary monday-style-steps--content-on-top"
    id=""
  >
    <div
      className="monday-style-steps-header monday-style-steps-header--gallery"
    >
      <button
        aria-disabled={true}
        className="monday-style-steps-command monday-style-steps-command--backward monday-style-button monday-style-button--size-small monday-style-button--kind-tertiary monday-style-button--color-on-primary-color"
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        type="button"
      >
        Back
      </button>
      <div
        className="monday-style-steps-header_gallery"
        role="group"
      />
      <button
        aria-disabled={false}
        className="monday-style-steps-command monday-style-steps-command--forward monday-style-button monday-style-button--size-small monday-style-button--kind-primary monday-style-button--color-on-primary-color"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        type="button"
      >
        Next
      </button>
    </div>
  </div>
</div>
`;

exports[`Tipseen wizard tests Snapshot Tests renders correctly without title 1`] = `
<div
  className="monday-style-tipseen-basic-content monday-style-tipseen-wizard"
>
  <div
    className="monday-style-steps monday-style-tipseen-wizard_wizard monday-style-steps--on-primary monday-style-steps--content-on-top"
    id=""
  >
    <div>
      Popover message will appear here loremipsum dolor samet…
    </div>
    <div
      className="monday-style-steps-header monday-style-steps-header--gallery"
    >
      <button
        aria-disabled={true}
        className="monday-style-steps-command monday-style-steps-command--backward monday-style-button monday-style-button--size-small monday-style-button--kind-tertiary monday-style-button--color-on-primary-color"
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        type="button"
      >
        Back
      </button>
      <div
        className="monday-style-steps-header_gallery"
        role="group"
      >
        <button
          aria-current="step"
          aria-label="Step number 0"
          className="monday-style-steps-header_dot monday-style-steps-header_dot--is-active"
          onClick={[Function]}
          type="button"
        />
        <button
          aria-current={false}
          aria-label="Step number 1"
          className="monday-style-steps-header_dot"
          onClick={[Function]}
          type="button"
        />
        <button
          aria-current={false}
          aria-label="Step number 2"
          className="monday-style-steps-header_dot"
          onClick={[Function]}
          type="button"
        />
      </div>
      <button
        aria-disabled={false}
        className="monday-style-steps-command monday-style-steps-command--forward monday-style-button monday-style-button--size-small monday-style-button--kind-primary monday-style-button--color-on-primary-color"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        type="button"
      >
        Next
      </button>
    </div>
  </div>
</div>
`;
